pkg/kubelet/prober/prober.go

```diff
@@ -207,7 +207,7 @@ func extractPort(param intstr.IntOrString, container v1.Container) (int, error)
 	case intstr.Int:
 		port = param.IntValue()
 	case intstr.String:
-		if port, err = findPortByName(container, param.StrVal); err != nil {
+		if port, err = findPort(container, param.StrVal, v1.ProtocolTCP); err != nil {
```

Comment:
nit: the `extractPort` function is currently only used for tcp protocol but is isn't called out in the name. Or param should be leveled up to the `extractPort` function